plugins/test/dynamic_test.cc

```diff
@@ -178,7 +178,12 @@ void DynamicTest::TestBody() {
     CheckPhaseResults("request_headers", invoke.result(), stream, res);
   }
   if (cfg_.request_body_size() > 0) {
-    ADD_FAILURE() << "BODY processing not implemented yet";
+    for (auto& invoke : *cfg_.mutable_request_body()) {
+      auto res = stream.SendRequestBody(
+          std::move(*invoke.mutable_input()->release_content()));
```

Comment:
I think this is a slight memory leak: while it moves the content of the string, the std::string object itself is never freed. To explain this, suppose I expand the code to the following equivalent snippet (just giving the pointer a name):

```
std::string* body_str = *invoke.mutable_input()->release_content();
auto res = stream.SendRequestBody(std::move(*body_str));
```

While the payload of the data that `body_str` contains is moved out of `body_str`, `body_str` itself remains, and nothing frees it.

The simplest way to ensure that the std::string* is properly deleted is to wrap it in a unique_ptr:

```
std::unique_ptr<std::string> body_str = absl::WrapUnique(*invoke.mutable_input()->release_content());
auto res = stream.SendRequestBody(std::move(*body_str));
```

This way, the payload of `body_str` is moved, and unique_ptr ensures that `body_str` itself is deleted.